### PR TITLE
Master merge from develop

### LINKED ISSUE #4093: Y24-063 Create a new attribute retention_instruction in labware table and update the way how retention instructions are stored and retrieved
**User story**

As a developer, I need to update where retention instructions for labware are retrieved from and stored to.

- [x] Create a new attribute `retention_instruction` in `labware` table to store retention instructions for plates and tubes. This attribute can only take the following values: `"Destroy after 2 years"`, `"Return to customer after 2 years"` and `"Long term storage"`.
- [x] Update the current codebase to store retention instructions in `labware` table. Currently, they are stored in `custom_metadata` with  `custom_metadata.key` equal to `retention_instructions`.
- [x] Update the current codebase to retrieve retention instructions from `labware` table and display retention instruction ~in **Metadata** section of the Labware user interface, along with metadata coming from `custom_metadata`. We will introduce a separate page in https://github.com/sanger/sequencescape/issues/3676 to let admin users edit retention instructions.~ in the fields at the top of the Labware page (see the figure below). We will introduce a separate page in https://github.com/sanger/sequencescape/issues/3676 to let admin users edit retention instructions

image

**Who are the primary contacts for this story**
Dasun, Katy

**Who is the nominated tester for UAT**

**Acceptance criteria**
To be considered successful the solution must allow:
- [x] All retrievals and storage of retention instructions of **new submissions** must occur from/to `labware` table, and not from/to `custom_metadata` table.

**Dependencies**
This story is blocked by the following dependencies:

References
This story has a non-blocking relationship with:
- https://github.com/sanger/sequencescape/issues/4003

**Additional context**

### PR DESCRIPTION
**Pull request by dasunpubudumal**
Closes #4093, #3676, #4095

#### Changes proposed in this pull request

- https://github.com/sanger/sequencescape/issues/4093
- https://github.com/sanger/sequencescape/issues/3676
- https://github.com/sanger/sequencescape/issues/4095
- https://github.com/sanger/sequencescape/issues/4114

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: .release-version [hunk 1]
```diff
@@ -1 +1 @@
-14.32.2
+14.33.2
```
Comment (KatyTaylor) [inline .release-version:1]: Doesn't really matter, but this should be 14.33.0, right?

Comment (dasunpubudumal) [reply .release-version:1]: Oh no. :|

Yeah I'll update it.

File: app/views/labware/_asset_summary.html.erb [hunk 2]
```diff
@@ -100,5 +100,17 @@
...
+    <th>Retention Instruction</th>
+    <!-- First, check if retention instructions are there. -->
+    <% if asset.retention_instruction.blank? %>
+        <td>Not specified</td>
+    <!-- If retention instructions are present, display them. -->
+    <%  else  %>
+      <td><%= find_retention_instruction_from_key(asset.retention_instruction.to_sym) %></td>
+    <%  end %>
+    <!-- There shouldn't be a situation where retention instructions are there in both custom_metadata -->
+    <!-- and in labware table's retention instruction attribute -->
```
Comment (KatyTaylor) [inline app/views/labware/_asset_summary.html.erb:113]: Is this comment still meant to be here, after removing the backwards compatible code around custom metadata?

**Review by KatyTaylor — APPROVED** [review]
Spotted a couple of things, but approving anyway as they are not important.

**Pull request merged**